Command/DocumentCommand.php

```diff
@@ -0,0 +1,208 @@
+                InputOption::VALUE_OPTIONAL,
+                'Size of the elasticsearch bulk request',
+                500
+            )
+            ->addOption(
+                'raw',
+                null,
+                InputOption::VALUE_NONE,
+                'The content will be imported as is. Without any field validation, data stripping or field protection'
+            )
+            ->addOption(
+                'dont-sign-data',
+                null,
+                InputOption::VALUE_NONE,
+                'The content will not be signed during the import process'
+            )
+            ->addOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Also treat document in draft mode'
+            )
+            ->addOption(
+                'dont-finalize',
+                null,
+                InputOption::VALUE_NONE,
+                'Don\'t finalize document'
+            )
+            ->addOption(
+                'businessKey',
+                null,
+                InputOption::VALUE_NONE,
+                'Try to identify documents by their business keys'
+            );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        $arguments = array_values($input->getArguments());
+        array_shift($arguments);
+        list($contentTypeName, $archiveFilename) = $arguments;
+
+        $this->io->title('Make documents');
+        $this->io->section('Checking input');
+
+
+        $contentType = $this->contentTypeService->getByName($contentTypeName);
+        if (!$contentType instanceof ContentType) {
+            $this->io->error(sprintf('Content type %s not found', $contentTypeName));
+            return;
+        }
+
+        if ($contentType->getDirty()) {
+            $this->io->error(sprintf('Content type %s is dirty. Please clean it first', $contentTypeName));
+            return;
+        }
+        $this->contentType = $contentType;
+
+        if (!file_exists($archiveFilename)) {
+            $this->io->error(sprintf('Archive file %s does not exist', $archiveFilename));
+            return;
+        }
+        $this->archiveFilename = $archiveFilename;
+        $this->ready = true;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (!$this->ready) {
+            return -1;
+        }
+
+        $options = array_values($input->getOptions());
+        list($bulkSize, $rawImport, $dontSignData, $force, $dontFinalize, $replaceBusinessKey) = $options;
+
+        $signData = !$dontSignData;
+        $finalize = !$dontFinalize;
+
+        $this->io->section(sprintf('Start importing %s from %s', $this->contentType->getPluralName(), $this->archiveFilename));
+
+        $archive = new Archive();
```

Comment:
The Command should be seen as a Controller. Its the glue between the logic. It should not contain logic. From line 160 - 202 should not be hosted in the Command file.  Because it is executing "business" logic, it should be in a Service. When using Controllers and Commands, a Service is never overkill. A service makes the code reusable between commands and controllers, and are the tool to keep your those two "thin".

Reply:
It is good to have this Archive object, but a service should create it and handle the logic from line 160 - 202